pkg/detectors/godaddy/v1/godaddy.go

```diff
@@ -0,0 +1,145 @@
+package godaddy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+type GoDaddyEnv string
+
+var (
+	// ensure the scanner satisfies the interface at compile time.
+	_ detectors.Detector  = (*Scanner)(nil)
+	_ detectors.Versioner = (*Scanner)(nil)
+
+	defaultClient = common.SaneHttpClient()
+
+	// the key for the GoDaddy OTE environment is a 37-character alphanumeric string that may include underscores.
+	keyPattern = regexp.MustCompile(detectors.PrefixRegex([]string{"godaddy", "ote"}) + common.BuildRegex("a-zA-Z0-9", "_", 37))
+	// the secret for the GoDaddy OTE environment is a 22-character alphanumeric string.
+	secretPattern = regexp.MustCompile(detectors.PrefixRegex([]string{"godaddy", "ote"}) + common.BuildRegex("a-zA-Z0-9", "", 22))
+)
+
+// supported environments for godaddy
+var (
+	OTE  GoDaddyEnv = "api.ote-godaddy.com"
+	Prod GoDaddyEnv = "api.godaddy.com"
+)
+
+func (s *Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+
+	return defaultClient
+}
+
+func (s *Scanner) Version() int { return 1 }
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"godaddy"}
```

Comment:
question: I noticed that `keyPattern` and `secretPattern` include `ote` as a keyword. Should we also include it here?

Reply:
Yes, we can! Initially, I thought that if we find the `godaddy` keyword in the chunk, we could then look for the `ote` or `godaddy` prefix. However, including `ote` as a keyword directly won’t cause any harm.